test_plans/atomic_ref.asciidoc

```diff
@@ -0,0 +1,480 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for atomic_ref
+
+This is a test plan for the APIs described in SYCL 2020 section 4.15.3. Atomic references
+
+== Testing scope
+
+The test plan covers non-atomic behavior, the latter will be covered in a separate test plan
+
+=== Device coverage
+
+All of the tests described below are performed on the default device that is selected on the CTS command line
+
+== Type coverage
+
+All of the tests described below are performed using each of the
+following types as the `ReferencedType`.
+
+In Regular mode:
+
+* `int`
+* `float`
+
+In full conformance mode:
+
+* `int`
+* `unsigned int`
+* `long`
+* `unsigned long`
+* `float`
+
+In addition, if the device has `aspect::atomic64`, the following types are tested:
+
+* `long long`
+* `unsigned long long`
+* `double`
+
+Additionaly run same tests for `ReferencedType*`.
+
+Note: using T as `ReferencedType` or `ReferencedType*`.
+
+== Tests
+
+All non static tests run on following cases:
+
+* single_task with referenced value is accessor element - should be skipped for `address_space::local_space`
+* parallel_for with nd_range parameter and referenced value local_accessor element - should be skipped for `address_space::global_space`
+
+Note: all tests run with different values of `DefaultOrder` and `DefaultScope` template parameters, for unsupported values tests are skiped. To check if the special values of `DefaultOrder` parameters are supported should use `info::device::atomic_memory_order_capabilities`, `memory_order::relaxed` should be supported on all devices. To check if the special values for `DefaultScope` parameter are supported should use `info::device::atomic_memory_scope_capabilities`.
```

Comment:
I don't know which formatting standards are used in the repo or whether they exist at all, but it would be great if we had a certain limit on a line width: something like 80 or 120 symbols to improve readability of those documents in console and make it easier to view diff of any changes.